datalad/distribution/publish.py

```diff
@@ -15,108 +15,109 @@
+    generally you can publish to any sibling added to a dataset.
+    Publishing may or may not include subdatasets.
+    By default file handles are published without their actual content,
+    but they can be published including the content, of course.
 
-    This is basic implementation for testing purposes
     """
+    # TODO: Figure out, how to tell about tracking branch/upstream
+    #      (and the respective remote)
+    #      - it is used, when no destination is given
+    #      - it is configured to be the given destination, if there was no
+    #        upstream set up before, so you can use just "datalad publish" next
+    #        time.
+
+    # TODO: Doc!
 
     _params_ = dict(
         dataset=Parameter(
             args=("-d", "--dataset"),
-            doc="""specify the dataset to perform the publish operation on.  If
-            no dataset is given, an attempt is made to identify the dataset
-            based on the current working directory and/or the `path` given""",
+            metavar='DATASET',
+            doc="""specify the dataset to publish. If no dataset is given, an
+            attempt is made to identify the dataset based on the current
+            working directory""",
             constraints=EnsureDataset() | EnsureNone()),
-        dest=Parameter(
-            args=("dest",),
-            doc="""url, local path, or sibling name identifying the publication
-            target""",
-            nargs="?",
-            constraints=EnsureStr() | EnsureNone()),
-        path=Parameter(
-            args=("path",),
-            doc="path/name of the dataset component to publish",
-            nargs="*",
+        to=Parameter(
+            args=("--to",),
+            metavar='DEST',
+            doc="""sibling name identifying the publication target. If no
+            destination is given an attempt is made to identify the target based
+            on the dataset's configuration.""",
+            # TODO: See TODO at top of class!
             constraints=EnsureStr() | EnsureNone()),
         recursive=Parameter(
             args=("-r", "--recursive"),
             action="store_true",
-            doc="recursively publish all components of the dataset"),
-        with_data=Parameter(
-            args=("--with-data",),
-            doc="shell pattern",
+            doc="recursively publish all subdatasets of the dataset"),
+        path=Parameter(
+            args=("path",),
+            metavar='PATH',
+            doc="path(s), that may point to file handle(s) to publish including "
+                "their actual content or to subdataset(s) to be published. If a "
+                "file handle is published with its data, this implicitly means "
+                "to also publish the (sub)dataset it belongs to.",
             constraints=EnsureStr() | EnsureNone(),
-            nargs='*'),)
+            nargs='*'),
+        to_annex=Parameter(
+            args=("--to-annex",),
```

Comment:
"--to" prefix is somewhat suboptimal here, since somewhat collides with "--to" and in general -- shouldn't it be `--annex-copy-opts` or smth like that?  happen we decide to engage/expose some other additional annex command, we would be ready to add passing options to it as well, that is why imho better to make this one explicitly for 'copy'